src/runge_kutta.jl

```diff
@@ -0,0 +1,509 @@
+                                  2//27   0      0       0         0          0        0        0      0       0     0 0 0
+                                  1//36   1//12  0       0         0          0        0        0      0       0     0 0 0
+                                  1//24   0      1//8    0         0          0        0        0      0       0     0 0 0
+                                  5//12   0    -25//16  25//16     0          0        0        0      0       0     0 0 0
+                                  1//20   0      0       1//4      1//5       0        0        0      0       0     0 0 0
+                                -25//108  0      0     125//108  -65//27    125//54    0        0      0       0     0 0 0
+                                 31//300  0      0       0       61//225    -2//9     13//900   0      0       0     0 0 0
+                                  2       0      0     -53//6    704//45   -107//9    67//90    3      0       0     0 0 0
+                                -91//108  0      0      23//108 -976//135   311//54  -19//60   17//6  -1//12   0     0 0 0
+                               2383//4100 0      0    -341//164 4496//1025 -301//82 2133//4100 45//82 45//164 18//41 0 0 0
+                                  3//205  0      0       0        0        -6//41     -3//205  -3//41  3//41   6//41 0 0 0
+                              -1777//4100 0      0    -341//164 4496//1025 -289//82 2193//4100 51//82 33//164 12//41 0 1 0],
+                              [41//840 0 0 0 0 34//105 9//35 9//35 9//280 9//280 41//840 0 0
+                               0 0 0 0 0 34//105 9//35 9//35 9//280 9//280 0     41//840 41//840],
+                               [0,    2//27, 1//9, 1//6 , 5//12, 1//2 , 5//6 , 1//6 , 2//3 , 1//3 , 1 , 0, 1]
+                            )
+
+
+###########################
+# Runge-Kutta solvers
+# (Hairer & Wanner 1992 p.134, p.165-169)
+
+function make_consistent_types(fn, y0, tspan, btab)
+    # There are a few types involved which somehow need to be consistent:
+    #
+    # Et = eltype(tspan)
+    # Ey = eltype(y0)
+    # Ef = eltype(Tf)
+    #
+    # There are also the types of the containers, but they are not
+    # needed as `similar` is used to make containers.
+    # Tt = typeof(tspan)
+    # Ty = typeof(y0)              # note, this can be a scalar
+    # Tf = typeof(F(tspan(1),y0))  # note, this can be a scalar
+    #
+    # Returns
+    # - Et: eltype of time, needs to be a real "continuous" type, at
+    #       the moment a FloatingPoint
+    # - Ey: eltype of y
+    # - Ef: eltype of fn(t,y)
+    #   --> both of these are set to typeof(y0[1]/tspan[1])
+    # - btab: tableau with entries converted to Et
+
+    Ey = typeof(y0[1]/(tspan[end]-tspan[1]))
+    Ef = Ey # this could be changed if function type become available
+
+    Et = promote_type(eltype(tspan), Float16)
```

Comment:
There is definitely room for improvement there.  The idea is to make it a floating point type.  This would avoid errors for `Vector{Int}` tspans.  And I thought that this would do the right thing: promote most things to `Float64` but keep the `Float32` and `Float16` as they are.  But it doesn't, so it's just plain wrong.  I'll come up with something more clever ;-)